### PR TITLE
Examples for BOTS

### PR DESCRIPTION
**Pull request by ShenQianli**
## Description

Trinity version code for BOTS: A Unified Framework for Bayesian Online Task Selection in LLM Reinforcement Finetuning.

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

**Comment by HYLcool** [comment]
Several modifications are made in this PR as well:
1. Remove SELECTOR_METRIC from pipeline logs to avoid unsupported type error.
2. Call the to_dict method of EID in the to_dict method of Experience to avoid serialization error.
3. Support `nested_query`: it allows users to specify the task_desc and truth in the second or deeper levels with the '.' operator. For example, setting `response_key` to `reward_model.ground_truth` is equivalent to fetching the response from `raw_task["reward_model"]["ground_truth"]`

File: trinity/common/workflows/workflow.py [hunk 1]
```diff
@@ -64,13 +81,13 @@ def to_workflow(
     @property
     def task_desc(self) -> Union[str, None]:
         prompt_key = self.format_args.prompt_key
-        return self.raw_task[prompt_key] if prompt_key in self.raw_task else None  # type: ignore
+        return nested_query(prompt_key, self.raw_task)  # type: ignore
```
Comment (pan-x-c) [inline trinity/common/workflows/workflow.py]: Deprecated property, don't change or use

Move the code into `BOTSMathBoxedWorkflow` would be better

File: trinity/common/workflows/workflow.py [hunk 2]
```diff
@@ -64,13 +81,13 @@ def to_workflow(
...
 
     # Deprecated property, will be removed in the future
     @property
     def truth(self) -> Union[str, None]:
         response_key = self.format_args.response_key
-        return self.raw_task[response_key] if response_key in self.raw_task else None  # type: ignore
+        return nested_query(response_key, self.raw_task)
```
Comment (pan-x-c) [inline trinity/common/workflows/workflow.py]: Same as above

File: examples/bots/workflow/bots_math_boxed_reward.py
Comment (pan-x-c) [inline examples/bots/workflow/bots_math_boxed_reward.py:1]: rename the directory from `plugins` -> `workflow`

File: README.md [hunk 3]
```diff
@@ -82,6 +82,7 @@ Trinity-RFT is a flexible, general-purpose framework for reinforcement fine-tuni
 
 ## 🚀 News
 
+* [2025-11] Introducing [BOTS](https://github.com/modelscope/Trinity-RFT/tree/main/examples/bots): dynamic RL task selection for efficient LLM fine-tuning ([paper](https://arxiv.org/pdf/2510.26374)).
```
Comment (pan-x-c) [inline README.md]: Update `README_zh`

Comment (yanxi-chen) [reply README.md]: Also, would be nice to add a line in the "Advanced data pipelines" row of "Tutorials and Guidelines" table. There are 4 places to apply this update: `readme.md`, `readme_zh.md`, and two `main.md` files in `docs`.

File: examples/bots/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+Remember to modify the model/data path in `bots.yaml` and `random.yaml` accordingly.
+
+##### Step 3: Training
+Launch training by executing:
+```bash
+trinity run --config examples/bots/bots.yaml --plugin-dir examples/bots/plugins
+```
+The improvement over random selection baseline can be stably obtained 🤖🤖🤖.
+
+<img src="https://img.alicdn.com/imgextra/i3/O1CN016wQqpG1wFq00KWzV7_!!6000000006279-2-tps-1894-1066.png" alt="Agentic workflows" width="700" />
```
Comment (yanxi-chen) [inline examples/bots/README.md]: Please specify what is the X-axis

Comment (yanxi-chen) [reply examples/bots/README.md]: Otherwise lgtm

File: README_zh.md [hunk 5]
```diff
@@ -67,13 +67,13 @@ Trinity-RFT 是一个灵活、通用的大语言模型（LLM）强化微调（RF
...
+| 数据流水线进阶能力 | + [Rollout 任务混合与选取](https://modelscope.github.io/Trinity-RFT/en/main/tutorial/develop_selector.html)<br>+ [在线任务课程](https://github.com/modelscope/Trinity-RFT/tree/main/examples/bots) ([论文](https://arxiv.org/pdf/2510.26374))<br>+ [经验回放](https://github.com/modelscope/Trinity-RFT/tree/main/examples/ppo_countdown_exp_replay)<br>+ [高级数据处理能力 &  Human-in-the-loop](https://modelscope.github.io/Trinity-RFT/en/main/tutorial/example_data_functionalities.html)                                                                                                                                                               |
```
Comment (pan-x-c) [inline README_zh.md]: 在线任务课程 -> 在线任务选择

File: docs/sphinx_doc/source_zh/main.md [hunk 6]
```diff
@@ -48,13 +48,13 @@ Trinity-RFT 是一个灵活、通用的大语言模型（LLM）强化微调（RF
...
+| 多轮智能体场景 | + [拼接多轮任务](/tutorial/example_multi_turn.md)<br>+ [通用多轮任务](/tutorial/example_step_wise.md)<br>+ [调用智能体框架中的 ReAct 工作流](/tutorial/example_react.md)                                                                                                                                                                                                                                                                                                                                                                                                                                |
+| 数据流水线进阶能力 | + [Rollout 任务混合与选取](/tutorial/develop_selector.md)<br>+ [在线任务课程](https://github.com/modelscope/Trinity-RFT/tree/main/examples/bots) ([论文](https://arxiv.org/pdf/2510.26374))<br>+ [经验回放](https://github.com/modelscope/Trinity-RFT/tree/main/examples/ppo_countdown_exp_replay)<br>+ [高级数据处理能力 &  Human-in-the-loop](/tutorial/example_data_functionalities.md)                                                                                                                                                                                                                 |
```
Comment (pan-x-c) [inline docs/sphinx_doc/source_zh/main.md]: -> 在线任务选择

File: examples/bots/README_zh.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+```bash
+trinity run --config examples/bots/bots.yaml --plugin-dir examples/bots/workflow
+```
+相比随机选择基线的提升可以被稳定地观察到🤖🤖🤖.
+
+<img src="https://gw.alicdn.com/imgextra/i2/O1CN0127XIYA1FHBgkXCKQ5_!!6000000000461-2-tps-947-533.png" alt="Agentic workflows" width="700" />
+
+### 完整复现
+
+想要完整复线我们论文中的结果，请从[这里](https://dail-wlcb.oss-cn-wulanchabu.aliyuncs.com/public/BOTS_verl_version.zip)下载verl版本的框架。
```
Comment (pan-x-c) [inline examples/bots/README_zh.md]: -> 复现

**Comment by ShenQianli** [comment]
> Consider add a cross-link at
> 
> ```
> * 📊 面向数据工程师。[[教程]](https://modelscope.github.io/Trinity-RFT/zh/main/tutorial/develop_operator.html)
>   - 设计针对任务定制的数据集，构建处理流水线以支持数据清洗、增强以及人类参与场景
>   - 示例：[数据处理基础](https://modelscope.github.io/Trinity-RFT/zh/main/tutorial/example_data_functionalities.html) [在线任务选择](xxx)
> ```

Done @yxdyc.
